src/backend/parser/analyze.c

```diff
@@ -1911,6 +1921,16 @@ transformSetOperationStmt(ParseState *pstate, SelectStmt *stmt)
 
 	/* add jnsitem to column namespace only */
 	addNSItemToQuery(pstate, jnsitem, false, false, true);
+	
+	/* tsql needs the leftmost query's targetlist and ns to handle ORDER BY */
+	sv_targetList = qry->targetList;
```

Comment:
We should add a hook type named :  pre_transfrom_sort_clause_hook, and move those code into tsql  ( and also the value of set_op_ns_stack into babel